src/com/ms/silverking/cloud/zookeeper/ZooKeeperExtended.java

```diff
@@ -42,1024 +43,1081 @@
+    Map<String, String> stringData;
+
+    stringData = new HashMap<>();
+    for (Map.Entry<String, byte[]> e : data.entrySet()) {
+      stringData.put(e.getKey(), new String(e.getValue()));
     }
-    
-    public Stat getStat(String path) throws KeeperException {
-        try { 
-            Stat    stat;
-            
-            stat = new Stat();
-            getData(path, false, stat);
-            return stat;
-        } catch (InterruptedException ie) {
-            throw new RuntimeException("panic");
-        }
+    return stringData;
+  }
+
+  private Map<String, Integer> convertToIntData(Map<String, byte[]> data) {
+    Map<String, Integer> intData;
+
+    intData = new HashMap<>();
+    for (Map.Entry<String, byte[]> e : data.entrySet()) {
+      intData.put(e.getKey(), NumConversion.bytesToInt(e.getValue()));
     }
+    return intData;
+  }
+
+  public byte[] getByteArray(String path) throws KeeperException {
+    return getByteArray(path, null);
+  }
+
+  public byte[] getByteArray(String path, Watcher watcher) throws KeeperException {
+    return getByteArray(path, watcher, new Stat());
+  }
 
-    public ChildrenCallbackOp getChildrenAsync(String path, boolean watch) throws KeeperException {
-        ChildrenCallbackOp  op;
-        
-        op = newChildrenCallbackOp();   
-        super.getChildren(path, watch, (ChildrenCallback)this, op.getRequestUUID());
-        return op;
+  public byte[] getByteArray(String path, Watcher watcher, Stat stat) throws KeeperException {
+    try {
+      byte[] data;
+
+      data = getData(path, watcher, stat);
+      return data;
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
+  }
+
+  public String getString(String path) throws KeeperException {
+    byte[] data;
+
+    data = getByteArray(path);
+    return new String(data);
+  }
+
+  public String getString(String path, Watcher watcher, Stat stat) throws KeeperException {
+    byte[] data;
+
+    data = getByteArray(path, watcher, stat);
+    return new String(data);
+  }
+
+  public int getInteger(String path) throws KeeperException {
+    byte[] data;
+
+    data = getByteArray(path);
+    return NumConversion.bytesToInt(data);
+  }
 
-    public ChildrenCallbackOp getChildrenAsync(String path) throws KeeperException {
-        return getChildrenAsync(path, false);
+  public boolean exists(String path) throws KeeperException {
+    try {
+      return super.exists(path, false) != null;
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
-    
-    public void close() {
-        try {
-            super.close();
-        } catch (InterruptedException ie) {
-            throw new RuntimeException("panic");
-        }
+  }
+
+  public List<String> getChildren(String path, Watcher watcher) throws KeeperException {
+    try {
+      return super.getChildren(path, watcher);
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)